### PR TITLE
Update the installation assumptions doc

### PR DESCRIPTION
**Pull request by webknjaz**
Improve installation doc

#### PR Type

- Docs Pull Request

### CODE REVIEW

File: doc/source/installation.rst [hunk 1]
```diff
@@ -56,7 +56,13 @@ available on your controller or host machines.
 
       $ pip install --upgrade --user setuptools
 
-.. _setuptools: https://pypi.org/project/setuptools/
+  .. _setuptools: https://pypi.org/project/setuptools/
+
+.. warning::
+
+  Pip v19 series has an isolation bug of setuptools being exposed to the
+  package build env. That is why it's highly recommended to upgrade user
```
Comment (decentral1se) [inline doc/source/installation.rst]:
```suggestion
  package build environment. That is why it's highly recommended to upgrade the user
```

File: doc/source/installation.rst [hunk 2]
```diff
@@ -56,7 +56,13 @@ available on your controller or host machines.
 
       $ pip install --upgrade --user setuptools
 
-.. _setuptools: https://pypi.org/project/setuptools/
+  .. _setuptools: https://pypi.org/project/setuptools/
+
+.. warning::
+
+  Pip v19 series has an isolation bug of setuptools being exposed to the
```
Comment (decentral1se) [inline doc/source/installation.rst]: Reference?

File: doc/source/installation.rst [hunk 3]
```diff
@@ -121,9 +127,27 @@ Please see :ref:`docker-usage-example` for usage.
...
-in "`development mode`_" so that new updates can be obtained by simply doing a
-``git pull`` in the repository's directory.
+Due to the rapid pace of development on this tool, you might want to
+install and update a bleeding-edge version of Molecule from Git.
+
+Follow the instructions below to do the initial install and subsequent
+updates.
+
+The package distribution that you'll get installed will be autogenerated
+and will contain a commit hash information making it easier to refer to
```
Comment (decentral1se) [inline doc/source/installation.rst]:
```suggestion
and will contain commit hash information making it easier to refer to
```

File: doc/source/installation.rst [hunk 4]
```diff
@@ -121,9 +127,27 @@ Please see :ref:`docker-usage-example` for usage.
...
+certain unstable version should the need to send a bug report arise.
+
+.. warning::
+
+  Please avoid using ``--editable``/``-e`` `development mode`_ when
+  installing Molecule with Pip. This not very well supported and only
+  needed when doing development.
+  For contributing purposes, it's already integrated with ``tox``
```
Comment (decentral1se) [inline doc/source/installation.rst]: Perhaps `For contributing purposes, you can rely on the Tox command line interface. Please see ... <link to our tox command docs>`?

File: doc/source/installation.rst [hunk 5]
```diff
@@ -121,9 +127,27 @@ Please see :ref:`docker-usage-example` for usage.
...
+  properly.
+
+  .. _`development mode`:
+     https://setuptools.readthedocs.io/en/latest\
```
Comment (decentral1se) [inline doc/source/installation.rst]: Can be an intersphinx link instead please.

File: doc/source/installation.rst (hunk 4 above)
Comment (webknjaz) [reply doc/source/installation.rst]: Yeah, this should probably be improved...

File: doc/source/installation.rst (hunk 5 above)
Comment (webknjaz) [reply doc/source/installation.rst]: It'd be an unrelated change. This was in the original doc version. I'd like to keep only related changes in this PR.

File: doc/source/installation.rst (hunk 2 above)
Comment (webknjaz) [reply doc/source/installation.rst]: I'll add it

**Pull request merged**